File: thru-hiker-log-app/src/app/models/trail.ts
import { IEntry } from './entry';

export interface ITrail {
  id?: number;
  name: string;
  location: string;
  length: number;
  lengthType: string;
  startDate: Date;
  endDate?: Date;
  gearListUrl: string;
  notes: string;
  entries?: IEntry[];
}

File: ThruHikerLogApi/ThruHikerLogApi/Models/Trail.cs
namespace ThruHikerLogApi.Models
{
    public class Trail
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public double Length { get; set; }
        public string LengthType { get; set; } // e.g., "miles" or "nights"
        public string? Location { get; set; }
        public string? Description { get; set; }
        public string? GearListUrl { get; set; }
        public List<Entry>? Entries { get; set; }
    }
}
